Mathlib/Algebra/Group/Pi/Sum.lean

```diff
@@ -0,0 +1,110 @@
+
+@[to_additive] instance instElimOne [∀ (i : ια), One (α i)] [∀ i : ιβ, One (β i)]
+    (i : ια ⊕ ιβ) : One (Sum.elim α β i) :=
+  match i with
+  | Sum.inl i => inferInstanceAs (One (α i))
+  | Sum.inr i => inferInstanceAs (One (β i))
+
+@[to_additive] instance instElimMul [∀ (i : ια), Mul (α i)]
+    [∀ i : ιβ, Mul (β i)] (i : ια ⊕ ιβ) : Mul (Sum.elim α β i) :=
+  match i with
+  | Sum.inl i => inferInstanceAs (Mul (α i))
+  | Sum.inr i => inferInstanceAs (Mul (β i))
+
+@[to_additive] instance instElimMulOneClass [∀ (i : ια), MulOneClass (α i)]
+    [∀ i : ιβ, MulOneClass (β i)] (i : ια ⊕ ιβ) : MulOneClass (Sum.elim α β i) :=
+  { one_mul := fun _ => match i with
+    | Sum.inl i => one_mul (M := α i) _
+    | Sum.inr i => one_mul (M := β i) _
+    mul_one := fun _ => match i with
+    | Sum.inl i => mul_one (M := α i) _
+    | Sum.inr i => mul_one (M := β i) _ }
+
+@[to_additive] instance instElimSemigroup [∀ (i : ια), Semigroup (α i)]
+    [∀ i : ιβ, Semigroup (β i)] (i : ια ⊕ ιβ) : Semigroup (Sum.elim α β i) :=
+  { mul_assoc := fun _ _ _ => match i with
+    | Sum.inl i => mul_assoc (G := α i) _ _ _
+    | Sum.inr i => mul_assoc (G := β i) _ _ _ }
+
+@[to_additive] instance instElimCommSemigroup [∀ (i : ια), CommSemigroup (α i)]
+    [∀ i : ιβ, CommSemigroup (β i)] (i : ια ⊕ ιβ) : CommSemigroup (Sum.elim α β i) :=
+  { mul_comm := fun _ _ => match i with
+    | Sum.inl i => mul_comm (G := α i) _ _
+    | Sum.inr i => mul_comm (G := β i) _ _ }
+
+@[to_additive] instance instElimPow {γ : Type*} [∀ i : ια, Pow γ (α i)]
+    [∀ i : ιβ, Pow γ (β i)] (i : ια ⊕ ιβ) :
+    Pow γ (Sum.elim α β i) :=
+  match i with
+  | Sum.inl i => inferInstanceAs (Pow γ (α i))
+  | Sum.inr i => inferInstanceAs (Pow γ (β i))
+
+@[to_additive] instance instElimMonoid [∀ (i : ια), Monoid (α i)]
+    [∀ i : ιβ, Monoid (β i)] (i : ια ⊕ ιβ) : Monoid (Sum.elim α β i) :=
+  { instElimSemigroup i, instElimMulOneClass i with
+    npow := npowRec }
+
+@[to_additive] instance instElimCommMonoid [∀ (i : ια), CommMonoid (α i)]
+    [∀ i : ιβ, CommMonoid (β i)] (i : ια ⊕ ιβ) : CommMonoid (Sum.elim α β i) :=
+  { mul_comm := CommSemigroup.mul_comm }
+
+@[to_additive] instance instElimInv [∀ (i : ια), Inv (α i)]
+    [∀ i : ιβ, Inv (β i)] (i : ια ⊕ ιβ) : Inv (Sum.elim α β i) :=
+  match i with
+  | Sum.inl i => inferInstanceAs (Inv (α i))
+  | Sum.inr i => inferInstanceAs (Inv (β i))
+
+@[to_additive] instance instElimDiv [∀ (i : ια), Div (α i)]
+    [∀ i : ιβ, Div (β i)] (i : ια ⊕ ιβ) : Div (Sum.elim α β i) :=
+  match i with
+  | Sum.inl i => inferInstanceAs (Div (α i))
+  | Sum.inr i => inferInstanceAs (Div (β i))
+
+@[to_additive] instance instElimDivInvMonoid [∀ (i : ια), DivInvMonoid (α i)]
+    [∀ i : ιβ, DivInvMonoid (β i)] (i : ια ⊕ ιβ) : DivInvMonoid (Sum.elim α β i) :=
+  { div_eq_mul_inv := fun _ _ => match i with
+    | Sum.inl i => div_eq_mul_inv (G := α i) _ _
+    | Sum.inr i => div_eq_mul_inv (G := β i) _ _
+    zpow := zpowRec }
```

Comment:
This creates an instance diamond with the `instElimPow` above; you should be able to set `zpow x z := x ^ z` or similar and deal with the fallout.

Reply:
Whoops, thank you! I have done this and the other suggestions.